src/Mouf/Database/Patcher/DatabasePatchInstaller2.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ Copyright (C) 2013 David Négrier - THE CODING MACHINE
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+namespace Mouf\Database\Patcher;
+
+use Doctrine\DBAL\Connection;
+use Mouf\ClassProxy;
+use Mouf\Installer\PackageInstallerInterface;
+use Mouf\InstanceProxy;
+use Mouf\MoufInstanceNotFoundException;
+use Mouf\MoufManager;
+use Mouf\UniqueIdService;
+
+/**
+ * This utility class is in charge of registering a new database patch in the patch system.
+ * This class is designed to be used in a package installer, for instance when a package needs to create
+ * tables in database.
+ *
+ * @author David Negrier <david@mouf-php.com>
+ */
+class DatabasePatchInstaller2 implements PackageInstallerInterface
+{
+    /**
+     * (non-PHPdoc)
+     * @see \Mouf\Installer\PackageInstallerInterface::install()
+     * @param MoufManager $moufManager
+     * @throws \Mouf\MoufException
+     */
+    public static function install(MoufManager $moufManager) {
+        // Let's get the database connection (i.e. the instance 'dbalConnection')
+        $dbConnection = $moufManager->get('dbalConnection');
+        $dbConnectionDescriptor = $moufManager->getInstanceDescriptor('dbalConnection');
+
+        //Let's get the instance of PatchConnection or create it if not exist
+        try {
+            $patchConnection = $moufManager->get('patchConnection');
```

Comment:
You can use this instead:

https://github.com/thecodingmachine/security.simplelogincontroller/blob/3.1/src/Mouf/Security/SimpleLoginInstaller.php#L37